### ISSUE TITLE
Build error with latest source and OpenDNS.

### ISSUE THREAD

**Issue by gawth** [opendnssec/SoftHSMv2#299]
Hi, I am trying to build SoftHSMv2 on Centos 7.
I have a fresh build of the latest OpenSSL installed
I have run `./configure --disable-gost --disable-non-paged-memory`

But when I try to make I eventually get the following error...
```
/bin/sh ../../../libtool  --tag=CXX   --mode=link g++  -g -O2 -Wall -Wextra -std=c++11 -fvisibility=hidden   -o softhsm2-keyconv softhsm2-keyconv.o base64.o softhsm2-keyconv-ossl.o ../../lib/crypto/OSSLComp.o  -L/usr/local/lib -lcrypto -ldl
libtool: link: g++ -g -O2 -Wall -Wextra -std=c++11 -fvisibility=hidden -o softhsm2-keyconv softhsm2-keyconv.o base64.o softhsm2-keyconv-ossl.o ../../lib/crypto/OSSLComp.o  -L/usr/local/lib -lcrypto -ldl
softhsm2-keyconv-ossl.o: In function `save_rsa_pkcs8(char*, char*, key_material_t*)':
/home/gawth/dev/SoftHSMv2/src/bin/keyconv/softhsm2-keyconv-ossl.cpp:107: undefined reference to `RSA_set0_factors'
/home/gawth/dev/SoftHSMv2/src/bin/keyconv/softhsm2-keyconv-ossl.cpp:108: undefined reference to `RSA_set0_crt_params'
/home/gawth/dev/SoftHSMv2/src/bin/keyconv/softhsm2-keyconv-ossl.cpp:109: undefined reference to `RSA_set0_key'
softhsm2-keyconv-ossl.o: In function `save_dsa_pkcs8(char*, char*, key_material_t*)':
/home/gawth/dev/SoftHSMv2/src/bin/keyconv/softhsm2-keyconv-ossl.cpp:201: undefined reference to `DSA_set0_pqg'
/home/gawth/dev/SoftHSMv2/src/bin/keyconv/softhsm2-keyconv-ossl.cpp:202: undefined reference to `DSA_set0_key'
softhsm2-keyconv-ossl.o: In function `crypto_init()':
/home/gawth/dev/SoftHSMv2/src/bin/keyconv/softhsm2-keyconv-ossl.cpp:54: undefined reference to `OPENSSL_init_crypto'
collect2: error: ld returned 1 exit status
```

Any suggestions?  
Thanks,
Alan

**Comment by bellgrim** [comment]
Cannot recreate this issue. Do you can more detail, e.g. version of SoftHSM and OpenSSL?

Also a little bit odd that you get an undefined reference to OPENSSL_init_crypto when the code is calling OpenSSL_add_all_algorithms()

**Comment by gawth** [comment]
Thanks for the speedy response.  

I realised that I had an old version of openssl in my path.  I have since updated path and openssl now reports itself as "OpenSSL 1.1.1-dev" - i.e. the latest build from the git repository.

However, that did not resolve the issue.  I am going to close the issue as I have gone with a binary install for now.  I am sure the compilation issue will turn out to be a config problem or something.

Thanks,
Alan
